src/sql/workbench/browser/editor/profiler/profilerInput.ts

```diff
@@ -156,6 +158,10 @@ export class ProfilerInput extends EditorInput implements IProfilerSession {
 		}
 	}
 
+	public get isXELFileSession(): boolean {
```

Comment:
What about just `isFileSession`? @kisantia this is supposed to be generic across all providers, right? So core shouldn't have any logic/references special to XEL if we can avoid it.

Reply:
I think this is only meant to be used for profiler, but correct me if this isn't the case @SakshiS-harma. Ideally all this profiler and load XEL file code would be in the profiler extension, but almost all of the profiler code is in core and the extension has very minimal code and it isn't worth it at the moment to move all of it into the extension.

Reply:
I can update it to isFileSession, but even for the other spots - they are really only meant for profiler code, not actually generic. And as Kim mentioned, profiler extension just has one dialog and some related UI stuff, but most of the profiler code sits within core (which ideally should be in its own extension, but for now not worth the effort).

Reply:
Yeah, so I guess I assumed that with the profiler feature living in the extensibility layer it was supposed to be a general thing that any provider could implement and use - and they'd just define what a profiler "session" looked like.

I have no idea what the state of that is though - so maybe that was never even a goal in the first place. Maybe something worth talking to @kburtram about. Probably not a blocker, although it doesn't hurt to try and keep it generic where we can.

Reply:
I will update it to be generic!